integration/setup/teardown.go

```diff
@@ -57,12 +64,90 @@ func teardown(ctx context.Context, config Config) error {
 	return nil
 }
 
-func deleteHostPeerVPC(config Config) error {
-	log.Printf("Deleting Host Peer VPC stack")
+func ensureDeletedStack(ctx context.Context, config Config, stackName string, wait bool) error {
+	config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring deletion of stack %#q", stackName))
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding out stack %#q", stackName))
+
+		_, err := getStack(ctx, config, stackName)
+		if IsNotFound(err) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find stack %#q", stackName))
+			return nil
+
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found stack %#q", stackName))
+	}
+
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("triggering stack %#q deletion", stackName))
+
+		err := startStackDeletion(ctx, config, stackName)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("triggered stack %#q deletion", stackName))
+	}
+
+	if wait {
```

Comment:
This can't be false right now, what is this useful for?

Reply:
The whole point of the PR is to make the thing synchronous. Otherwise it deletes VPC stack, and proceeds but the stack still exists (DELETION_IN_PROGRESS) which prevents from creating the stack with the same name.